exonum-java-binding-common/src/main/java/com/exonum/binding/common/serialization/UInt32Serializer.java

```diff
@@ -62,7 +63,7 @@ public Integer fromBytes(byte[] serializedValue) {
       int x;
       if ((x = serializedValue[pos++]) >= 0) {
         return x;
-      } else if (serializedValue.length - pos < 9) {
+      } else if (serializedValue.length - pos < serializedValue.length - 1) {
```

Comment:
I am not sure what this condition means. It used to be `9`, but turned into (**a reminder to everyone: the type of the serializer _changed_ from `sint32` to `uint32`**)
```
serializedValue.length - pos < serializedValue.length - 1
  Strip duplicates:
- pos < - 1
  Keep going
1 < pos
  Swap
pos > 1
```

Is `pos > 1` the intended condition (please check my math)?

Reply:
`9` came from the original code and it means `MAX_BYTES -1`.  Such as we need to remove unused space tail in the buffer we can't use `MAX_BYTES` here.
Yes, I'll simplify this statement.

Reply:
> Such as we need to remove unused space tail in the buffer we can't use MAX_BYTES here.

Not sure I understand — do we remove anything? When this condition can possibly be true (we only `break` out of the block, never `continue`)? What the original code of uint32 serializer uses?